packages/blade/src/components/Box/BaseBox/BaseBox.web.tsx

```diff
@@ -1,7 +1,13 @@
+import React from 'react';
 import styled from 'styled-components';
+import { getBaseBoxStyles, getDependencyProps } from './getBaseBoxStyles';
 import type { BaseBoxProps } from './types';
-import getBaseBoxStyles from './getBaseBoxStyles';
 
-const BaseBox = styled.div<BaseBoxProps>(getBaseBoxStyles);
+const BaseBox = styled.div<BaseBoxProps>((props) => {
+  const dependencyProp = getDependencyProps(props);
```

Comment:
We're calculating `dependencyProp` on all renders it seems? 🤔 

This is doing `JSON.stringify` which could potentially be an expensive operation to be run for all renders _(if I'm reading it right)_.

Reply:
I think JSON.stringify would get expensive if objects are too nested? we don't have nested props here. Just straight value or responsive object. It's also comparitively a lot less expensive than our getBaseBoxStyles for most cases-

<img width="160" alt="image" src="https://user-images.githubusercontent.com/30949385/220101855-6c08cdb0-f1a1-4008-9565-8035da76fc66.png">

:react-god: also seems to suggest JSON.stringify (I believe dependency prop comparison is faster at lot of places if it's just string)
<img width="590" alt="image" src="https://user-images.githubusercontent.com/30949385/220102379-d3588a1d-10fe-44fe-8c80-e766b63a008b.png">

Reply:
Twitter Source: https://twitter.com/dan_abramov/status/1104414272753487872?s=20

Reply:
Btw did you see any improvements with memoizing? I see the numbers you shared above but I think they're all with memoizing.

Reply:
Yup that getBaseBoxStyles doesn't run when you change screen size now or in-between. There is a difference in number of times the function runs.




https://user-images.githubusercontent.com/30949385/220506025-e50d64b3-59a2-4ef7-b024-9a3da0cf87c9.mov


https://user-images.githubusercontent.com/30949385/220506045-a0340931-16e1-47c2-adcb-8a9049a138ef.mov


without memo (these are avg numbers and currentPerf object has current run numbers)
<img width="250" alt="image" src="https://user-images.githubusercontent.com/30949385/220506929-13241aee-789d-4f00-a8dd-1b9b20b5d942.png">

with memo
<img width="322" alt="image" src="https://user-images.githubusercontent.com/30949385/220507113-5699f0e1-c684-4583-aa5b-5769206eb63e.png">


Not major difference in time but it would help since Box might be used at a lot of places.

Reply:
Nice, if I'm reading this correctly then it's quite good improvement from avg `0.25` to `0.03` ish 😸 

Thanks for digging in.

Reply:
> without memo (these are avg numbers and currentPerf object has current run numbers)

How are you checking the perf here? 

It might be worth checking the perf with react-dev-tool's profiler for concrete numbers.